internal/worker/caasapplicationprovisioner/application.go

```diff
@@ -185,6 +185,7 @@ func (a *appWorker) loop() error {
 	}
 
 	var done bool
+	var ready bool
```

Comment:
Maybe block the two vars together with the ones below.

Maybe a comment that ready is set to true as soon as the app is recorded as being alive, but that we may still get other events up to that point; and that as soon as the app goes to dying, ready is set to false.